Commit ID: 760e5a604ece31d8635703b5a0ab520813b140bf
Change ID: ttzlzoykusmrxzlruxonywvkvkulnsox
Author   : Lauri Tirkkonen <lauri@hacktheplanet.fi> (2026-01-04 18:31:37)
Committer: Lauri Tirkkonen <lauri@hacktheplanet.fi> (2026-01-04 18:43:29)

    management & signal: disable pprof listen

    these clash with each other when run in the same netns. since neither
    one is configurable and since pprof is only useful for profiling,
    disable the pprof listening on both services until the issue is resolved
    upstream.

    Link: https://github.com/netbirdio/netbird/issues/4923

diff --git a/management/main.go b/management/main.go
index ff8482f972..8db5437078 100644
--- a/management/main.go
+++ b/management/main.go
@@ -1,19 +1,12 @@
 package main
 
 import (
-	"log"
-	"net/http"
-	// nolint:gosec
-	_ "net/http/pprof"
 	"os"
 
 	"github.com/netbirdio/netbird/management/cmd"
 )
 
 func main() {
-	go func() {
-		log.Println(http.ListenAndServe("localhost:6060", nil))
-	}()
 	if err := cmd.Execute(); err != nil {
 		os.Exit(1)
 	}
diff --git a/signal/cmd/run.go b/signal/cmd/run.go
index bf8f8e3271..1ddf3c7299 100644
--- a/signal/cmd/run.go
+++ b/signal/cmd/run.go
@@ -92,8 +92,6 @@
 		RunE: func(cmd *cobra.Command, args []string) error {
 			flag.Parse()
 
-			startPprof()
-
 			opts, certManager, tlsConfig, err := getTLSConfigurations()
 			if err != nil {
 				return err
